### PR TITLE
Dataset API documentation

### LINKED ISSUE #302: Documentation for high-level API
The high-level is really great for working with datasets. I think we are nearing a stage, where a documentation is required, so that more people can use it.

### PR DESCRIPTION
**Pull request by rschwanhold**
### Description:
- `pdoc` is used to create the documentation. Running `docs/api.sh` opens the documentation in the browser.
- The documentation is currently limited to the Dataset API
 
### Issues:
- fixes #302 

### Todos:
Make sure to delete unnecessary points or to check all before merging:
 - [x] Updated Changelog

### CODE REVIEW

**Comment by rschwanhold** [comment]
What `pdoc` and `jinja2` versions are you using?

**Comment by normanrz** [comment]
`pdoc 6.6.0` and `Jinja2 3.0.0`

**Comment by rschwanhold** [comment]
There was a bug with `jinja2 < 3.0` and `pdoc`. This was fixed in a recent release. Even though I had the latest `pdoc` version, the `poetry.lock` still contained an old entry

**Review by normanrz — COMMENTED** [review]
I think this is a great starting point for a useful documentation! I added a few comments. I also added @jstriebel who may have some additional ideas.

In general, I personally learn the most about APIs from code examples in documentations. I think it would be a good idea to add short samples for key methods/classes. This isn't a blocker and can be done in a follow-up.

File: wkcuber/api/Dataset.py [hunk 1]
```diff
@@ -160,10 +184,12 @@ def create_with_properties(cls, properties: Properties) -> "AbstractDataset":
         # initialize object
         return cls(dataset_dir)
 
-    def get_properties(self) -> Properties:
-        return self.properties
-
     def get_layer(self, layer_name: str) -> LayerT:
+        """
+        Returns the layer called `layer_name` of this dataset. The return type is `wkcuber.api.Layer`.
```
Comment (jstriebel) [inline wkcuber/api/Dataset.py:189]:
```suggestion
        Returns the layer called `layer_name` of this dataset. The return type is `wkcuber.api.Layer.Layer`.
```

File: wkcuber/api/Dataset.py [hunk 2]
```diff
@@ -179,6 +205,17 @@ def add_layer(
...
         **kwargs: Any,
     ) -> LayerT:
+        """
+        Creates a new layer called `layer_name` and adds it to the dataset.
+        The dtype can either be specified per layer or per channel.
+        If neither of them are specified, `uint8` per channel is used as default.
+        When creating a `wkcuber.api.Layer.SegmentationLayer` (category="segmentation"),
+        the parameter `largest_segment_id` also has to be specified.
+
+        The return type is `wkcuber.api.Layer`.
```
Comment (jstriebel) [inline wkcuber/api/Dataset.py:215]:
```suggestion
        The return type is `wkcuber.api.Layer.Layer`.
```

File: wkcuber/api/Layer.py [hunk 3]
```diff
@@ -64,9 +59,15 @@
...
 class Layer(Generic[MagT]):
+    """
+    A `Layer` consists of multiple `wkcuber.api.MagDataset`s, which store the same data in different magnifications.
+    """
 
     COLOR_TYPE = "color"
     SEGMENTATION_TYPE = "segmentation"
+    """
+    There are two different types (`COLOR_TYPE`, `SEGMENTATION_TYPE`).
+    """
```
Comment (jstriebel) [inline wkcuber/api/Layer.py:70]: Not sure if that's useful here. Also, they are only used together with `Dataset.add_layer`, so they should be defined there rather.

```suggestion
```

File: wkcuber/api/Layer.py [hunk 4]
```diff
@@ -75,40 +76,75 @@ def __init__(
         dtype_per_channel: np.dtype,
         num_channels: int,
     ) -> None:
+        """
+        Creates the folder `name` in the directory if the `dataset`.
```
Comment (jstriebel) [inline wkcuber/api/Layer.py:80]:
```suggestion
        Creates the folder `name` in the directory of `dataset`.
```

File: wkcuber/api/Layer.py [hunk 5]
```diff
@@ -75,40 +76,75 @@ def __init__(
...
+
+        A `Layer` cannot exist without a dataset. The desired procedure to create a new layer for a dataset is to call
+        `wkcuber.api.Dataset.AbstractDataset.add_layer` instead of creating and then adding it manually.
```
Comment (jstriebel) [inline wkcuber/api/Layer.py:83]: :+1:

**Review by jstriebel — COMMENTED** [review]
Nice work! Similar to @normanrz suggestions I'd add some examples, but that might be part of another PR. I'v added some other remarks and questions in the comments. Besides those, I was wondering why you added the empty `__init__` files? Also I noticed the upper-case module-names, which also cause confusion in the docs (e.g. `wkcuber.api.Layer` vs `wkcuber.api.Layer.Layer`). I opened a separate issue for those, see #327.

File: wkcuber/api/Dataset.py [hunk 6]
```diff
@@ -160,10 +184,12 @@ def create_with_properties(cls, properties: Properties) -> "AbstractDataset":
         # initialize object
         return cls(dataset_dir)
 
-    def get_properties(self) -> Properties:
-        return self.properties
-
     def get_layer(self, layer_name: str) -> LayerT:
+        """
+        Returns the layer called `layer_name` of this dataset. The return type is `wkcuber.api.Layer.Layer.Layer`.
```
Comment (normanrz) [inline wkcuber/api/Dataset.py:189]:
```suggestion
        Returns the layer called `layer_name` of this dataset. The return type is `wkcuber.api.Layer.Layer`.
```

**Comment by rschwanhold** [comment]
I will add examples after the restructuring of the API.

**Review by jstriebel — APPROVED** [review]
LGTM for now :+1:

**Review by normanrz — APPROVED** [review]

**Pull request merged**